pkg/helpers/reader/reader.go

```diff
@@ -88,6 +89,11 @@ func (r *ResourceReader) applyOneObject(info *resource.Info) error {
 	helper := resource.NewHelper(info.Client, info.Mapping).
 		DryRun(r.dryRun)
 
+	modified, err := kubectlutil.GetModifiedConfiguration(info.Object, false, unstructured.UnstructuredJSONScheme)
```

Comment:
Hi @qiujian16 could you add some comments/tests for this, I do not understand why moving this func upper can fix the issue.

Reply:
this is to get the modified object, it has to be called before info.Get() otherwise the info.Object will be from the server. A test is added.